muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/workspacenotes/rest/WorkspaceNoteRESTService.java

```diff
@@ -0,0 +1,323 @@
+   * workspaceNote: "text here",
+   * workspaceEntityId: 123
+   * };
+   * 
+   * returns:
+   * 
+   * WorkspaceNoteRestModel: = { 
+   * id: 1,
+   * title: "title here"
+   * workspaceNote: "text here",
+   * workspaceEntityId: 123,
+   * owner: 1
+   * };
+   * 
+   * Errors:
+   * 400 Bad request if workspaceEntityId is null
+   */
+  
+  @POST
+  @Path("/workspacenote/")
+  @RESTPermit (handling = Handling.INLINE, requireLoggedIn = true)
+  public Response createWorkspaceNote(WorkspaceNoteRestModel workspaceNote) {
+    
+    if (workspaceNote.getWorkspaceEntityId() == null) {
+      return Response.status(Status.BAD_REQUEST).build();
+    }
+    
+    WorkspaceEntity workspaceEntity = workspaceEntityController.findWorkspaceEntityById(workspaceNote.getWorkspaceEntityId());
+    
+    if (workspaceEntity == null) {
+      return Response.status(Status.BAD_REQUEST).entity("Workspace entity not found").build();
+    }
+    
+    WorkspaceNote newWorkspaceNote = workspaceNoteController.createWorkspaceNote(sessionController.getLoggedUserEntity().getId(), workspaceNote.getTitle(), workspaceNote.getWorkspaceNote(), workspaceEntity.getId());
+
+    return Response.ok(toRestModel(newWorkspaceNote)).build();
+  }
+  
+  /*
+   * mApi() call mApi().workspacenotes.workspacenote.update(WorkspaceNoteRestModel) 
+   *  
+   *  Parameter rest model must contain owner, workspaceEntityId & nextSiblingId. 
+   *  
+   *  example:
+   *  {owner: 14, workspaceEntityId: 23, title: "Title", workspaceNote: "updatedNote", nextSiblingId: 2}
+   *  
+   *  
+   *  Editable fields are only title, workspaceNote & nextSiblingId
+   *  
+   *  returns updated rest model
+   *  
+   *  WorkspaceNoteRestModel: = { 
+   *    id: 1,
+   *    title: "updated title here",
+   *    workspaceNote: "updated text here",
+   *    workspaceEntityId: 123,
+   *    owner: 1
+   * };
+   *  
+   *  Errors:
+   *  404 Not found if workspaceNote not found
+   *  400 Bad request if userEntityId is null
+   *  403 Forbidden if userEntityId does not match with logged user
+   */
+  @PUT
+  @Path ("/workspacenote")
+  @RESTPermit (handling = Handling.INLINE, requireLoggedIn = true)
+  public Response updateWorkspaceNote(WorkspaceNoteRestModel restModel) {
+    
+    WorkspaceNote workspaceNote = workspaceNoteController.findWorkspaceNoteById(restModel.getId());
+    
+    if (workspaceNote == null) {
+      return Response.status(Status.NOT_FOUND).build();
+    }
+    
+    if (restModel.getOwner() == null) {
+      return Response.status(Status.BAD_REQUEST).build();
+    } 
+    
+    if(restModel.getWorkspaceEntityId() == null || !restModel.getWorkspaceEntityId().equals(workspaceNote.getWorkspace())) {
+      return Response.status(Status.BAD_REQUEST).entity("Workspace entity id mismatch").build();
+    }
+    
+    // users can only update their own notes (except admins)    
+    if (!restModel.getOwner().equals(sessionController.getLoggedUserEntity().getId())) {
```

Comment:
I assume this should also be checked against `workspaceNote.getOwner` to check if the entity can be modified? Or maybe check that `restModel.getOwner == workspaceNote.getOwner` (bad request) and that `workspaceNote.getOwner == loggedUser.id` (forbidden)?

Reply:
Just don't use `==` check with Long objects :)